tests/benchmark/spaces_benchmarks/bm_spaces.h

```diff
@@ -13,19 +13,25 @@
 #include "VecSim/spaces/L2_space.h"
 #include "VecSim/spaces/IP/IP.h"
 #include "VecSim/spaces/L2/L2.h"
+#include "VecSim/spaces/functions/AVX512.h"
+#include "VecSim/spaces/functions/AVX.h"
+#include "VecSim/spaces/functions/AVX512BW_VBMI2.h"
+#include "VecSim/spaces/functions/AVX2.h"
+#include "VecSim/spaces/functions/SSE3.h"
+#include "VecSim/spaces/functions/SSE.h"
 
 #include "bm_spaces_class.h"
 
 // Defining the generic benchmark flow: if there is support for the optimization, benchmark the
 // function.
-#define BENCHMARK_DISTANCE_F(type_prefix, arch, metric, bm_name)                                   \
+#define BENCHMARK_DISTANCE_F(type_prefix, arch, metric, bm_name, opt_supported)                    \
     BENCHMARK_DEFINE_F(BM_VecSimSpaces, type_prefix##_##arch##_##metric##_##bm_name)               \
     (benchmark::State & st) {                                                                      \
-        if (opt < spaces::ARCH_OPT_##arch) {                                                       \
+        if (!opt_supported) {                                                                      \
             st.SkipWithError("This benchmark requires " #arch ", which is not available");         \
```

Comment:
for consistency with "arch" param
```suggestion
#define BENCHMARK_DISTANCE_F(type_prefix, arch, metric, bm_name, arch_supported)                    \
    BENCHMARK_DEFINE_F(BM_VecSimSpaces, type_prefix##_##arch##_##metric##_##bm_name)               \
    (benchmark::State & st) {                                                                      \
        if (!arch_supported) {                                                                      \
            st.SkipWithError("This benchmark requires " #arch ", which is not available");         \
```